src/target_set.c

```diff
@@ -238,17 +421,75 @@ static int ts_invoke (const data_set_t *ds, value_list_t *vl, /* {{{ */
     return (-EINVAL);
   }
 
+  orig = *vl;
+
   if (data->meta != NULL)
   {
-    meta_data_clone_merge(&(vl->meta), data->meta);
+    char temp[DATA_MAX_NAME_LEN*2];
+    int meta_entries;
+    char **meta_toc;
+
+    if ((new_meta = meta_data_create()) == NULL)
+    {
+      ERROR ("Target `set': failed to create replacement metadata.");
+      return (-ENOMEM);
+    }
+
+    meta_entries = meta_data_toc (data->meta, &meta_toc);
+    for (int i = 0; i < meta_entries; i++)
+    {
+      const char *key = meta_toc[i];
+      char *string;
+      int status;
+
+      status = meta_data_get_string (data->meta, key, &string);
```

Comment:
In this case, `data->meta` is filled in by `ts_config_add_meta()`, which uses `meta_data_add_string()`, so it's guaranteed to be a string. Plus, the error printed by `meta_data_get_string()` will not show that this was a failure in `target_set`.

I really like the idea of `meta_data_as_string()`, and added it for use in `ts_subst()`.